src/toast/utils.py

```diff
@@ -36,6 +36,56 @@
 )
 
 
+def set_numba_openmp():
+    have_numba_omp = False
+    have_numba_tbb = False
+    try:
+        from numba.npyufunc import tbbpool
+
+        have_numba_tbb = True
+    except ImportError:
+        # no TBB
+        pass
+
+    try:
+        from numba.npyufunc import omppool
+
+        have_numba_omp = True
+    except ImportError:
+        # no OpenMP support
+        pass
+
+    try:
+        from numba import config, njit, threading_layer
+
+        if have_numba_omp:
+            config.THREADING_LAYER = "omp"
+        elif have_numba_tbb:
+            config.THREADING_LAYER = "tbb"
+        else:
+            config.THREADING_LAYER = "default"
+
+        # In order to get numba to actually select a threading layer, we must
+        # trigger compilation of a parallel function.
+        @njit(parallel=True)
+        def foo(a, b):
+            return a + b
+
+        x = np.arange(10.0)
+        y = x.copy()
+        foo(x, y)
```

Comment:
as we are running this anyway, we could as well throw an `assert` to make a quick test of the result.

Reply:
Ok, and I guess in that case it means that numba is broken and we should just exit the function